src/test/java/dog/pawbook/storage/JsonAdaptedProgramTest.java

```diff
@@ -0,0 +1,79 @@
+package dog.pawbook.storage;
+
+import static dog.pawbook.storage.JsonAdaptedOwner.MISSING_FIELD_MESSAGE_FORMAT;
+import static dog.pawbook.testutil.Assert.assertThrows;
+import static dog.pawbook.testutil.TypicalEntities.INDEPENDENCE_TRAINING;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.Test;
+
+import dog.pawbook.commons.exceptions.IllegalValueException;
+import dog.pawbook.model.managedentity.Name;
+import dog.pawbook.model.managedentity.program.Program;
+import dog.pawbook.model.managedentity.program.Session;
+import javafx.util.Pair;
+
+public class JsonAdaptedProgramTest {
+    private static final String INVALID_NAME = "@bedience Training`";
+    private static final String INVALID_SESSION = "111-12";
+    private static final String INVALID_DOGID = "  ";
+    private static final String INVALID_TAG = "#friend";
+
+    private static final Integer INDEPENDENCE_TRAINING_ID = 2;
+    private static final String VALID_NAME = INDEPENDENCE_TRAINING.getName().toString();
+    private static final List<JsonAdaptedSession> VALID_SESSIONS = INDEPENDENCE_TRAINING.getSessions().stream()
+        .map(JsonAdaptedSession::new)
+        .collect(Collectors.toList());
```

Comment:
is the indentation here wrong? I think it should be 8 spaces more than the previous line right?